File: other_languages/python/counting_sort.py


from shared_sort import int_array


def counting_sort(array):
    counter = [0 for x in range(max(array) + 1)]
    sorted_array = []

    for index, value in enumerate(array):
        counter[value] += 1

    for index, value in enumerate(counter):
        while value > 0:
            sorted_array.append(index)
            value -= 1
    return sorted_array


def main(array):
    counter = [0 for x in range(max(array) + 1)]
    print(f"Original array {array}")
    for index, value in enumerate(array):
        counter[value] += 1
    print(f"Map of array: {counter}")
    print(f"Sorted array: {counting_sort(array)}")


if __name__ == '__main__':
    main(int_array)

File: other_languages/javascript/countingSort.js


const int_array = [1, 3, 5, 8, 12, 90, 33, 1, 5, 76, 32];


function countingSort (array) {
    const max = Math.max(...array) + 1;
    let counter = new Array(max).fill(0);
    let sorted_array = [];

    for (let i = 0; i < array.length; i++) {
        counter[array[i]]++;
    }

    for (let i = 0; i < max; i++) {
        while (counter[i] > 0) {
            sorted_array.push(i);
            counter[i]--;
        }
    }

    return sorted_array;
}

console.log(int_array);
const answer = countingSort(int_array);
console.log(answer);
